### PR TITLE
AirfRANS: Fourier+4L/256d+no-EMA long training (full 180-min budget)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

PR #2478 achieved val_primary/surface_mse = 0.2387 in only 8 epochs on a 30-min budget. The training curve was still steeply descending at epoch 8. With the full 180-min budget and --epochs 999, we expect 40-50 epochs of training — potentially pushing into the 0.10-0.15 range. This is the most critical AirfRANS experiment right now: we already know the config works, we just need more time.

## Instructions

Run the exact winning configuration from PR #2478 but with the full 180-min budget:

```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 150 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --wandb_group hinata-airfrans-fourier-4L256d-longrun
```

**CRITICAL:** `--epochs 999` is required — the default in train.py is 2 epochs, which will stop training prematurely regardless of SENPAI_MAX_EPOCHS.

## Baseline

- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Current best (val):** 0.2387
- **Current best (test):** 0.2079
- **full_val/volume_mse:** 0.2933
- **Best PR:** #2478 (senku — Fourier + 4L/256d + no-EMA, 8 epochs, AdamW lr=5e-4, T_max=150)
- **Baseline W&B run:** vwb9teqa
- **Reproduce command:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 5e-4 --cosine_t_max 150 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999`

## Report

Please report:
- val_primary/surface_mse (per epoch)
- test_primary/surface_mse (final)
- full_val/volume_mse
- Per-channel surface MSE breakdown: Ux, Uy, p, nut
- Total epochs completed
- Convergence trajectory (still improving at cutoff?)

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2497 because Superseded: hinata reassigned to #2567 (AirfRANS T_max sweep).

**Comment by tcapelle** [comment]
STUDENT hinata:

## Results

### Summary
The long training run dramatically validated the hypothesis — **89 epochs** of training (vs 8 in the baseline) yielded a **95.2% improvement** in val surface MSE and **93.4% in test surface MSE**. The model was still improving at cutoff (epoch 89 was the best epoch).

### Key Metrics

| Metric | Baseline (#2478) | This Run | Change |
|--------|-----------------|----------|--------|
| **val_primary/surface_mse** | 0.2387 | **0.0115** | **-95.2%** |
| **test_primary/surface_mse** | 0.2079 | **0.0137** | **-93.4%** |
| **full_val/volume_mse** | 0.2933 | **0.0985** | **-66.4%** |
| **full_test/volume_mse** | — | 0.1074 | — |

### Per-Channel Surface MSE (Test)

| Channel | Value |
|---------|-------|
| Ux | 3.11e-5 |
| Uy | 3.38e-6 |
| p | 0.0547 |
| nut | 2.53e-6 |

### Per-Channel Surface MSE (Val, Best Epoch 89)

| Channel | Value |
|---------|-------|
| Ux | 3.40e-5 |
| Uy | 3.43e-6 |
| p | 0.0461 |
| nut | 2.42e-6 |

### Training Details

- **Epochs completed:** 89 (180-min timeout)
- **Best val epoch:** 89 (still improving at cutoff!)
- **W&B run:** r0ccctht
- **Peak GPU:** NVIDIA RTX PRO 6000 (GPU 0)
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 5e-4 --cosine-t-max 150 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --agent hinata --wandb-name "hinata/airfrans-fourier-4L256d-longrun" --wandb-group hinata-airfrans-fourier-4L256d-longrun`

### What Happened

The hypothesis was correct — the Fourier+4L/256d+no-EMA config with AdamW was severely epoch-starved at 8 epochs. With the full 180-min budget:

1. **Cosine LR cycling drives optimization:** With T_max=150 (300-step full cycle) and 360 steps/epoch, the LR oscillates ~1.2 times per epoch. This creates a characteristic pattern of val metric oscillation (troughs at LR minima, peaks at LR maxima) with progressively deeper troughs.

2. **Trough progression:** The best-epoch surface MSE improved dramatically across cosine cycles: 0.2500 (ep4) → 0.2036 (ep14) → 0.0613 (ep23) → 0.0388 (ep32) → 0.0202 (ep49) → 0.0159 (ep51) → 0.0145 (ep67) → **0.0115 (ep89)**

3. **Pressure channel dominates:** Surface MSE is almost entirely driven by the pressure (p) channel (0.046 at best epoch vs ~1e-5 for velocity components). The 95% improvement comes from learning better pressure representations.

4. **Volume metrics also improved massively:** Volume MSE dropped from 0.293 to 0.099 (66% improvement), with volume pressure MSE at 0.017.

5. **Still improving at cutoff:** Epoch 89 was the best epoch for both surface and volume MSE. The training curve had not plateaued — more epochs would likely yield further gains.

### Suggested Follow-ups

1. **Even longer training:** The model was still improving at 89 epochs. A 360-min budget could push below 0.01.
2. **T_max tuning:** T_max=150 creates wide LR oscillations. A larger T_max (e.g., 300-500) might reduce oscillation amplitude and improve convergence stability in later epochs.
3. **Cosine annealing with warm restarts:** The current cosine schedule repeats indefinitely. A schedule with decaying restarts could help converge more stably.
4. **Learning rate reduction:** Starting lr=5e-4 causes large oscillations in late training. A lower LR or LR decay could help.
